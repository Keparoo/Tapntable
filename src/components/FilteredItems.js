import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchItemsFromAPI } from '../actions/items';
import { Link as RouterLink } from 'react-router-dom';

import Spinner from '../components/Spinner';
import { clearCurrentCheck } from '../actions/currentCheck';
import { useHistory } from 'react-router-dom';
import {
  Container,
  List,
  ListItem,
  ListItemText,
  Paper,
  Typography,
  Link
} from '@mui/material';

/*  Render page with list of items in db where isActive=true
 *
 *   On mount, renders list of all items in API
 *
 *   Routed to /items
*/

const ItemList = ({ items }) => {
  console.debug('ItemList');

  // const items = useSelector((st) => st.items);
  // const user = useSelector((st) => st.user);
  // const dispatch = useDispatch();
  // const history = useHistory();
  // const [ isLoading, setIsLoading ] = useState(true);

  // if (!user.pin) history.push('/');

  // useEffect(
  //   () => {
  //     console.debug('ItemList useEffect on Mount');

  //     dispatch(clearCurrentCheck());
  //     async function fetchItem() {
  //       await dispatch(fetchItemsFromAPI());
  //       setIsLoading(false);
  //     }
  //     if (isLoading) {
  //       fetchItem();
  //     }
  //   },
  //   [ dispatch, isLoading ]
  // );

  // if (isLoading) return <Spinner />;

  // if (!isLoading && items.length === 0) {
  //   return <b>No items in database</b>;
  // }
  if (items.length === 0)
    return (
      <Container>
        <Paper elevation={3}>
          <Typography ml={2} mt={2} variant="h3" component="h1">
            No items match
          </Typography>
        </Paper>
      </Container>
    );

  return (
    <Container>
      <Paper elevation={3}>
        <Typography ml={2} mt={2} variant="h3" component="h1">
          Find item to set count
        </Typography>
        <List>
          {items.map((i) => (
            <Link
              key={i.id}
              to={`/items/${i.id}`}
              component={RouterLink}
              underline="none"
            >
              <ListItem>
                <ListItemText
                  primary={
                    <React.Fragment>
                      <strong>{i.name}</strong>, Id: {i.id}
                    </React.Fragment>
                  }
                  secondary={
                    <React.Fragment>
                      <strong>${i.price}</strong>, Category:{' '}
                      <strong>{i.category}</strong>, Destination:{' '}
                      <strong>{i.destination}</strong> Count:{' '}
                      <strong>{i.count || 'None'}</strong> isActive:{' '}
                      <strong>{i.isActive ? 'true' : 'false'}</strong> <br />
                      {i.description}
                    </React.Fragment>
                  }
                />
              </ListItem>
            </Link>
          ))}
        </List>
      </Paper>
    </Container>
  );
};

export default ItemList;
