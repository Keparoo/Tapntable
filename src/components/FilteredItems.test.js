import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import rootReducer from '../reducers/root';
import thunk from 'redux-thunk';
import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { MemoryRouter } from 'react-router-dom';
import FilteredItems from './FilteredItems';

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

const items = [];

it('renders without crashing', function() {
  render(
    <Provider store={store}>
      <MemoryRouter>
        <FilteredItems items={items} />
      </MemoryRouter>
    </Provider>
  );
});

it('matches snapshot', function() {
  const { asFragment } = render(
    <Provider store={store}>
      <MemoryRouter>
        <FilteredItems items={items} />
      </MemoryRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
