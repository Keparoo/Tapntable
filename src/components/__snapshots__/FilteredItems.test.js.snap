// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-11jlvvb-MuiTypography-root"
      >
        No matching items
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
